test/src/compiler/docker-gcc.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+set -e
+
+s2j_debian_image_tag="bullseye"
+s2j_gcc_command="gcc"
+s2j_gcc_version="10"
+s2j_mounts=()
+gcc_args=()
+
+while [ -n "$1" ]; do
+    case "$1" in
+        "-s2j-debian-image-tag")
+            shift
+            s2j_debian_image_tag="$1"
+            ;;
+        "-s2j-gcc-command")
+            shift
+            s2j_gcc_command="$1"
+            ;;
+        "-s2j-gcc-version")
+            shift
+            s2j_gcc_version="$1"
+            ;;
+        "-s2j-mount")
+            shift
+            s2j_mounts+=("--volume=$1:$1")
+            ;;
+        *)
+            gcc_args+=("$1")
+            ;;
+    esac
+    shift 1
+done
+
+gcc_image_name="sio2jail-compiler-${s2j_debian_image_tag}-${s2j_gcc_version}"
+
+docker build \
+    --tag "${gcc_image_name}:latest" \
```

Comment:
nit: arguably, `sio2jail-compiler:${s2j_debian_image_tag}-${s2j_gcc_version}` would be more appropriate (version in version, not in image name)